internal/isoeditor/rhcos.go

```diff
@@ -92,6 +100,65 @@ func (e *rhcosEditor) addIgnitionArchive(ignition string) error {
 	return ioutil.WriteFile(e.isoHandler.ExtractedPath("images/ignition.img"), archiveBytes, 0644)
 }
 
+func (e *rhcosEditor) addCustomRAMDisk(staticIPConfig string) error {
+	configPath := "/etc/static_ips_config.csv"
+	scriptPath := "/usr/lib/dracut/hooks/initqueue/settled/90-assisted-static-ip-config.sh"
+	scriptContent := constants.ConfigStaticIpsScript
+	imagePath := "/images/assisted_installer_custom.img"
+
+	f, err := os.Create(e.isoHandler.ExtractedPath(imagePath))
+	if err != nil {
+		return err
+	}
+
+	w := cpio.NewWriter(f)
+	if err = addFileToArchive(w, configPath, staticIPConfig, 0o664); err != nil {
+		return err
+	}
+	if err = addFileToArchive(w, scriptPath, scriptContent, 0o755); err != nil {
+		return err
+	}
+	if err = w.Close(); err != nil {
+		return err
+	}
+
+	// edit config to add new image to initrd
+	err = editFile(e.isoHandler.ExtractedPath("EFI/redhat/grub.cfg"), `(?m)^(\s+initrd) (.+| )+$`, fmt.Sprintf("$1 $2 %s", imagePath))
+	if err != nil {
+		return err
+	}
+	return editFile(e.isoHandler.ExtractedPath("isolinux/isolinux.cfg"), `(?m)^(\s+append.*initrd=\S+) (.*)$`, fmt.Sprintf("${1},%s ${2}", imagePath))
+}
+
+func addFileToArchive(w *cpio.Writer, path string, content string, mode cpio.FileMode) error {
+	// add all the directories in path
+	for dir := filepath.Dir(path); dir != "" && dir != "/"; dir = filepath.Dir(dir) {
+		hdr := &cpio.Header{
+			Name: dir,
+			Mode: 040755,
```

Comment:
isn't it similar to 0o755, or do we need it specifically for a directory?

Reply:
Yup, but this library needs me to specifically set the directory bit. ref: https://golang.org/pkg/os/#FileMode